Heroic-Games-Launcher/HeroicGamesLauncher
# [Feat] Auto Download Wine if no version was found in the system

## Description
This is the first feature I get from HyperPlay, this is the original PR:
https://github.com/HyperPlay-Gaming/hyperplay-desktop-client/pull/317

It basically checks if there is any Wine/Proton/Crossover available, or at least that it could found, if none was found it downloads the latest Wine-GE on Linux or the latest Wine-Crossover on macOS.
We wanted to implement this in Heroic but I decided to do it first in HyperPlay to check how we could get features and fixes from the fork.

The download happens during startup just like DXVK or when launching a game.
Since most version are around 80 to 100MB I think that should be fine for most people for now, but we can improve this after getting more feedback.

I also think this won't affect anyone on the Steam Deck or that uses Steam on linux to play.
It will affect more people on macOS that does not know what Wine or Crossover are and wants to just download and play a Windows game there.

It also fixes an issue on macOS that was causing games to fail to launch due to not finding the wine prefix.
Also changes the name of the Wine Manager to Compatibility Layer, since it also covers Proton that is not Wine itself.
Might also be a more friendly name.


## How to Test
To test this just make sure you rename all folders that have wine on your system like .steam, lutris folder, heroic/tools folder, etc. And check if Heroic will download the correct version on Startup and show it on the Compatibility layer screen.

---

Use the following Checklist if you have changed something on the Backend or Frontend:
